### ISSUE TITLE
Argument conflict on interface type fragments

### ISSUE THREAD

**Issue by eapache** [rmosolgo/graphql-ruby#418]
Currently it is not possible to pass different args to the same field depending on the concrete type. The following example results in an error about an argument conflict:
```graphql
foo {
  ... on TypeOne { image(maxWidth: 200) }
  ... on TypeTwo { image(maxWidth: 300) }
}
```

However, the spec (specifically the bottom of https://facebook.github.io/graphql/#sec-Field-Selection-Merging) gives an example very similar to this as a validly mergeable query since the two fields both cannot be encountered against the same object, so they are safe.

cc @dylanahsmith @swalkinshaw 

Also see https://github.com/graphql/graphql-js/issues/53 and https://github.com/graphql/graphql-js/pull/229

**Comment by cjoudrey** [comment]
I find [the current test](https://github.com/rmosolgo/graphql-ruby/blob/7e2a0ba3ae38937befa97273fff8b7f921e69067/spec/graphql/static_validation/rules/fields_will_merge_spec.rb) for this rule a bit hard to follow.

I'll probably take a stab at breaking it up into multiple tests using the examples provided in the spec.

I'll probably look into fixing this at the same time if nobody gets to it before me.

**Comment by rmosolgo** [comment]
BTW I'm checking out a fix for this now. Before executing the query, we build a tree of nodes that groups fields by each possible object type. I think if I do that grouping sooner, I could use the resulting tree for this validation. Roughly, for each node in tree, make sure all of its backing AST nodes have the same name and same arguments. It introduces an extra visit over that tree, but I don't think it can be helped! (I _think_ the graphql-js validation is also recursive, but I can't quite tell -- I can't nail down the recursive function chain.)

**Comment by rmosolgo** [comment]
I think this is fixed by #512, which will be released with 1.5.0! Please reopen if you still experience this bug with that version.
